lib/portage/tests/util/test_socks5.py

```diff
@@ -295,24 +297,36 @@ def testSocks5ServerAtExit(self):
                     "-c",
                     """
 import sys
+import threading
 
 from portage.const import PORTAGE_BIN_PATH
 from portage.util import socks5
 from portage.util._eventloop.global_event_loop import global_event_loop
 
 tempdir = sys.argv[0]
-loop = global_event_loop()
+threaded = bool(sys.argv[1])
 
 settings = {
     "PORTAGE_TMPDIR": tempdir,
     "PORTAGE_BIN_PATH": PORTAGE_BIN_PATH,
 }
 
-socks5.get_socks5_proxy(settings)
-loop.run_until_complete(socks5.proxy.ready())
-print(socks5.proxy._proc.pid, flush=True)
+def main():
+    loop = global_event_loop()
+    socks5.get_socks5_proxy(settings)
+    loop.run_until_complete(socks5.proxy.ready())
+    print(socks5.proxy._proc.pid, flush=True)
+
+if __name__ == "__main__":
+    if threaded:
+        t = threading.Thread(target=main)
+        t.start()
+        t.join()
```

Comment:
Note that functions like `get_socks5_proxy` doe not currently have internal locking for thread safety, but we can safely neglect that here since this test case only has a single thread interacting with the portage API at a given time.

We can consider adding locking in the future, but for now our main goal should be to simply allow the portage APIs to be used in a single worker thread when the main thread is busy running a UI.

Reply:
Even though we use weak references to loops inside our asycio wrapper module, we can be sure that the loop won't be garbage collected before the socks5 proxy is stopped, since the socks5 proxy references the loop indirectly via ForkProcess and [_GCProtector](https://github.com/gentoo/portage/pull/1284).